x-pack/plugins/monitoring/public/components/logstash/pipeline_viewer/models/graph/if_vertex.js

```diff
@@ -40,18 +41,36 @@ export class IfVertex extends Vertex {
     return this.outgoingEdges.find(e => e.when === true);
   }
 
+  get trueEdges() {
+    return this.outgoingEdges.filter(e => e.when);
+  }
+
   get falseEdge() {
     return this.outgoingEdges.find(e => e.when === false);
   }
 
+  get falseEdges() {
+    return this.outgoingEdges.filter(e => e.when === false);
+  }
+
   get trueOutgoingVertex() {
     return this.trueEdge ? this.trueEdge.to : null;
   }
 
+  get trueOutgoingVertices() {
+    const trueEdgeIds = this.trueEdges.map(e => e.to.id);
+    return getVerticesForIds(this.outgoingVertices, trueEdgeIds);
+  }
```

Comment:
Maybe I'm missing something here but, similar to how `trueOutgoingVertex` is implemented, couldn't you implement `trueOutgoingVertices` like so:

```js
return this.trueEdges.map(e => e.to);
```

Reply:
Thanks for simplifying that!